README.md

```diff
@@ -1,198 +1,233 @@
+- [Example by Specification](#example-by-specification) 
+- [Extra APIs](#extra-apis)
+  - [Location API](#location-api)
+  - [Search Archive API](#search-archive-api)
+  - [Account API](#account-api)
+- [Promise and Callback](#promise-and-callback)
+- [Coding Style](#coding-style)
+- [Run Regression](#run-regression)
+- [Error Management](#error-management)
+- [Changelog](#change-log)
+</details>
 
-## Requirement
+## Requirements
 
 - ES6 basic understanding
-- NodeJS coding skills
-- Node 7+ and NPM installed
+- Node 7+ and [NPM installed](https://www.npmjs.com/package/npm)
 
 ## Installation
 
-NPM 7+
-
 ```bash
 $ npm install google-search-results-nodejs
 ```
 
-[Link to npm package](https://www.npmjs.com/package/google-search-results-nodejs)
-
 ## Quick start
 
+The following example runs a search for `"coffee"` using your secret API key which you can find at [SerpApi Dashboard](https://serpapi.com/manage-api-key) page. 
+
+[Open in the online IDE](https://replit.com/@serpapi/google-search-results-nodejs-quick-start?v=1) (Replit).
+
 ```javascript
 const SerpApi = require('google-search-results-nodejs')
-const search = new SerpApi.GoogleSearch("Your Private Key")
+const search = new SerpApi.GoogleSearch("<your-serpapi-api-key>")
+
 search.json({
  q: "Coffee", 
  location: "Austin, TX"
 }, (result) => {
   console.log(result)
 })
  ```
-This example runs a search about "coffee" using your secret api key.
-
-The SerpApi service (backend)
- - searches on Google using the search: q = "coffee"
- - parses the messy HTML responses
- - return a standardizes JSON response
-The class GoogleSearch
- - Format the request to SerpApi server
- - Execute GET http request
- - Parse JSON into Ruby Hash using JSON standard library provided by Ruby
-Et voila..
-
-Alternatively, you can search:
-- Baidu using BaiduSearch class
-- Bing using BingSearch class
-- DuckDuckGo using DuckDuckGoSearch class
-- Yahoo using YahooSearch class
-- Ebay using EbaySearch class
-- Yandex using YandexSearch class
-- HomeDepot using HomeDepotSearch class
-- GoogleScholar using GoogleScholarSearch class
-- Youtube using YoutubeSearch class
-- Walmart using WalmartSearch
-- Apple App Store using AppleAppStoreSearch class
-- Naver using NaverSearch class
-
-See the [playground to generate your code.](https://serpapi.com/playground)
-
-## Example
- * [How to set SERP API key](#how-to-set-serp-api-key)
- * [Search API capability](#search-api-capability)
- * [Example by specification](#example-by-specification)
- * [Location API](#location-api)
- * [Search Archive API](#search-archive-api)
- * [Account API](#account-api)
- * [Promise and callback](#Promise-and-callback)
- * [Coding style](#coding-style)
- 
-### How to set SERP API key
-The SerpApi api_key can be set globally using a singleton pattern.
+
+### How SerpApi backend works
+
+![image](https://user-images.githubusercontent.com/78694043/192951614-81ea265f-a8d4-49eb-9b51-10b09868b875.png)
```

Comment:
I'm personally in love with your illustrations :hearts: 

![image](https://user-images.githubusercontent.com/282605/213178407-3a254b5d-e77f-4776-bc0c-414f2296aa88.png)

Reply:
Me too. 😍